msgraph/client.go

```diff
@@ -85,22 +78,25 @@ type Client struct {
 	DisableRetries bool
 
 	// RequestMiddlewares is a slice of functions that are called in order before a request is sent
-	RequestMiddlewares []RequestMiddleware
+	RequestMiddlewares *[]RequestMiddleware
```

Comment:
I think this wants to become:

```suggestion
	RequestMiddlewares []*RequestMiddleware
```

which'd mean the middleware functions themselves would be pointers?

Reply:
A slice of function pointers makes more sense 👍